### PR TITLE
bearer_token, bearer_token_file for Marathon SD

### PR DESCRIPTION
**Pull request by m-kraus**
Reflect the possibility to include bearer_token or bearer_token_file for Marathon SD

### CODE REVIEW

**Comment by beorn7** [comment]
This feature will only be available in the 1.6 release, won't it? If so, this should go to the `next-release` branch.

**Comment by m-kraus** [comment]
Ok, updated branch.

**Comment by grobie** [comment]
Please check the commit history now. You'll need to rebase your commit on top of the next-release branch.

**Comment by m-kraus** [comment]
Sorry for the mess... i hope it's correct now.

File: content/docs/operating/configuration.md [hunk 1]
```diff
@@ -662,6 +662,12 @@ See below for the configuration options for Marathon discovery:
 servers:
   - <string>
 
+# Optional bearer token authentication information.
+  [ bearer_token: <string> ]
```
Comment (grobie) [inline content/docs/operating/configuration.md]: The indentation doesn't look right here and below.

Comment (m-kraus) [inline content/docs/operating/configuration.md]: Yes I know, but I used the same indentation, that was needed in the target environment:
```
    marathon_sd_configs:
    - servers:
      - 'https://SERVER/marathon'
      bearer_token_file: token
      tls_config:
        insecure_skip_verify: true
```

Using other indentations neither bearer_token_file nor tls_config are used.

So maybe for clarity I should indent the comments the same way as the options?

Comment (grobie) [inline content/docs/operating/configuration.md]: This code snippet describes a single <marathon_sd_config>, not a list like your example. Your current indentation puts it under the `servers` key, which is both wrong and impossible.

**Comment by m-kraus** [comment]
Well then, adapted indentation

**Comment by grobie** [comment]
Thanks! One more thing, looking at the comments of bearer token in the main
<scrape_config> block, I think we should mirror the "It is mutually
exclusive with ..." parts.

On Wed, Mar 8, 2017, 17:03 Michael Kraus <notifications@github.com> wrote:

Well then, adapted indentation

—
You are receiving this because you commented.


Reply to this email directly, view it on GitHub
<https://github.com/prometheus/docs/pull/676#issuecomment-285152569>, or mute
the thread

File: content/docs/operating/configuration.md [hunk 2]
```diff
@@ -662,6 +662,14 @@ See below for the configuration options for Marathon discovery:
 servers:
   - <string>
 
+# Optional bearer token authentication information.
+# It is mutually exclusive with `bearer_token_file`.
+[ bearer_token: <string> ]
+
+# Optional bearer token file authentication information.
+# It is mutually exclusive with `bearer_token`.
+[ bearer_token_file: <filename>
```
Comment (grobie) [inline content/docs/operating/configuration.md]: Missing closing `]`.

**Comment by m-kraus** [comment]
Oh, well - not my day as it seems...

**Comment by grobie** [comment]
We've all been there ;)

**Pull request merged**